File: services/tradelog/tradelog-api-spec/src/main/java/com/example/tradelog/api/spec/model/TransactionType.java
package com.example.tradelog.api.spec.model;

public enum TransactionType {

    SHARE,
    SYNTHETIC_SHARE,
    OPTION,
    DIVIDEND,
    UNKNOWN;

    static public TransactionType lookup(String id) {
        try {
            return TransactionType.valueOf(id);
        } catch (IllegalArgumentException ex) {
            return UNKNOWN;
        }
    }
}

File: services/tradelog/tradelog-api-spec/src/main/kotlin/com/example/tradelog/api/spec/model/TransactionType.kt
package com.example.tradelog.api.spec.model

import java.lang.IllegalArgumentException

enum class TransactionType {
    SHARE,
    SYNTHETIC_SHARE,
    OPTION,
    DIVIDEND,
    UNKNOWN;

    companion object {
        fun lookup(id : String) : TransactionType {
            return try {
                valueOf(id)
            } catch (e : IllegalArgumentException) {
                UNKNOWN
            }
        }
    }

}
